tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -13984,7 +13984,13 @@ bool Sema::DiagnoseHLSLDecl(Declarator &D, DeclContext *DC, Expr *BitWidth,
     QualType eltQt(qt->getArrayElementTypeNoTypeQual(), 0);
     while (eltQt->isArrayType())
       eltQt = QualType(eltQt->getArrayElementTypeNoTypeQual(), 0);
-
+    if (hlslSource->IsWaveMatrixType(eltQt)) {
+      std::string typeName(
+          g_ArBasicTypeNames[hlslSource->GetTypeElementKind(eltQt)]);
```

Comment:
One last nit: does this need to be a std::string? that forces a memory allocation. It would be better if this could be a StringRef.

Reply:
Fixed.